### PR TITLE
More IP-XACT-like RegFieldDesc

### PR DESCRIPTION
**Pull request by mwachs5**
I didn't realize that IP-XACT supported finer grained explanations of `WriteType` (and less interestingly `ReadAction`). Add these, as well as `volatile`, and replace the overly vague `*SPECIAL` access Types.

Also update some descriptions in Bus Error Unit, Debug, and PLIC to use these and correct some other inaccuracies

### CODE REVIEW

File: src/main/scala/devices/tilelink/Plic.scala [hunk 1]
```diff
@@ -229,7 +231,8 @@ class TLPLIC(params: PLICParams, beatBytes: Int)(implicit p: Parameters) extends
     }
 
     def thresholdRegDesc(i: Int) = RegFieldDesc(s"threshold_$i", s"Interrupt & claim threshold for target $i. Maximum value is ${nPriorities}.",
-      reset=if (nPriorities > 0) None else Some(1), access=RegFieldAccessType.RWSPECIAL)
+      reset=if (nPriorities > 0) None else Some(1),
```
Comment (aswaterman) [inline src/main/scala/devices/tilelink/Plic.scala]: Threshold,  priority, and enable registers are _not_ reset.

Comment (mwachs5) [reply src/main/scala/devices/tilelink/Plic.scala]: right, but if there is only one priority value they are constant 1, right? That's what the code is trying to say.

Comment (aswaterman) [reply src/main/scala/devices/tilelink/Plic.scala]: oooh I misread the code. ya

File: src/main/scala/regmapper/RegField.scala [hunk 2]
```diff
@@ -7,34 +7,54 @@ import chisel3.util.{ReadyValidIO}
...
 ){
 }
 
-// Our descriptions are in terms of RegFields only, which is somewhat unusual for
-// developers who are used to things being defined as bitfields within registers.
-// The "Group" allows a string & (optional) description to be added which describes the conceptual "Group"
-// the RegField belongs to. This can be used by downstream flows as they see fit to
-// present the information.
+object RegFieldDescReserved {
+  def apply(): RegFieldDesc = RegFieldDesc("reserved", "", access=RegFieldAccessType.R, reset=Some(0))
```
Comment (hcook) [inline src/main/scala/regmapper/RegField.scala]: I would probably code this as a `def reserved: RegFieldDesc` method on an companion object named `RegFieldDesc`, especially if you think there might be more of them eventually

Comment (mwachs5) [reply src/main/scala/regmapper/RegField.scala]: 👍

**Pull request merged**